docs/tuplets.md

```diff
@@ -0,0 +1,123 @@
+
+## Standard Ratios
+There are certain ratios that are standard, like a triplet would usually have a ratio of 3:2. However, there are also non-standard ratios where you can use whatever you'd like, such as 8:5. It would be good to have the standard ratios defined so that, for instance, a user can create a triplet and not have to define the full ratio.
+
+There should only be a need to list the standard ratios for 2-9. In this case, we can just list them in `Tuplet` struct. This way, if the second number is not specific in the initializer and the first number is one of these, we can fill in the second number according to th static list of default ratios.
+
+## Compound Tuplets
+You can have a tuplet that is made up of another tuplet, which is made up of another tuplet, and so on. 
+
+In order to support this behavior, the `Tuplet` will store an array of type `NoteCollection` so that it can have a `Note` or `Tuplet` as a value in the array. Also, the `NoteCollection` will have to have a property or set of properties to communicate the duration and number of that duration for each `NoteCollection`. 
+    
+i.e. A 3:2 Eighth note tuplet will have a number of 2 and duration of `.eighth`. A single eighth note, would have a number of 1 and duration of `.eighth`
+
+## Validation
+A tuplet needs to always be full. This means if it is a 3:2 ratio, it needs to have 3 notes in it. However, the notes can be rests. Because of this, the only mutating function should be to replace a note, not remove or add.
+You may also have notes of different durations as part of the tuplet as seen [here](http://www2.siba.fi/muste1/index.php?id=100&la=en) where you have triplets with eighth notes and tied quarter notes.
+
+Because of these rules, some type of validation must be done in the initializer to ensure the given notes equate to a full tuplet. This must be built into the `Tuplet` struct.
+
+## API Design
+### API Definition
+**Open**: Naming is still up for debate. Having trouble with that.
+```swift
+struct Tuplet {
+    /// The notes that make up the tuplet
+    public private(set) var notes: [NoteCollection]
+    /// The number of notes of the specified duration that this tuplet contains
+    public let baseNoteCount: Int
+    /// The duration of the notes that define this tuplet
+    public let baseNoteDuration: NoteDuration
+    /// The number of notes that this tuplet fits in the space of
+    public let noteCountFit: Int
+    
+    init(_ count: Int, _ duration: NoteDuration, inSpaceOf baseCount: Int? = nil, _ baseDuration: NoteDuration? = nil, notes: [NoteCollection]) throws
+    mutating func replaceNote(at index: Int, with note: Note) throws
+}
+```
+### Secondary duration
+We can choose to allow a second duration to be specified like in [Finale](https://usermanuals.finalemusic.com/Finale2014Mac/Content/Finale/TPDLG.htm). However, in order to put the ratio on top, it seems like you need to convert to the same duration?
```

Comment:
Side note: In the link there is also "Appearance" and "Position" settings. Taking note here since it seems that this would imply that the render API would have to have a Tuplet class.